korge/src/commonMain/kotlin/com/soywiz/korge/view/filter/Convolute3Filter.kt

```diff
@@ -21,37 +21,37 @@ class Convolute3Filter(
 
         /** A Gaussian Blur Kernel. This [Matrix3D] can be used as [kernel] for [Convolute3Filter] */
         val KERNEL_GAUSSIAN_BLUR: Matrix3D = Matrix3D.fromRows3x3(
-            1f, 2f, 1f,
-            2f, 4f, 2f,
-            1f, 2f, 1f
-        ) * (1f / 16f)
+            1.0, 2.0, 1.0,
+            2.0, 4.0, 2.0,
+            1.0, 2.0, 1.0
+        ) * (1.0 / 16.0)
 
         /** A Box Blur Kernel. This [Matrix3D] can be used as [kernel] for [Convolute3Filter] */
         val KERNEL_BOX_BLUR: Matrix3D = Matrix3D.fromRows3x3(
-            1f, 1f, 1f,
-            1f, 1f, 1f,
-            1f, 1f, 1f
-        ) * (1f / 9f)
+            1.0, 1.0, 1.0,
+            1.0, 1.0, 1.0,
+            1.0, 1.0, 1.0
+        ) * (1.0 / 9.0)
 
         /** An Identity Kernel (doesn't perform any operation). This [Matrix3D] can be used as [kernel] for [Convolute3Filter] */
         val KERNEL_IDENTITY: Matrix3D = Matrix3D.fromRows3x3(
-            0f, 0f, 0f,
-            0f, 1f, 0f,
-            0f, 0f, 0f
+            0.0, 0.0, 0.0,
+            0.0, 1.0, 0.0,
+            0.0, 0.0, 0.0
         )
 
         /** An Edge Detection Kernel. This [Matrix3D] can be used as [kernel] for [Convolute3Filter] */
         val KERNEL_EDGE_DETECTION: Matrix3D = Matrix3D.fromRows3x3(
-            -1f, -1f, -1f,
-            -1f, +8f, -1f,
-            -1f, -1f, -1f
+            -1.0, -1.0, -1.0,
+            -1.0, +8.0, -1.0,
+            -1.0, -1.0, -1.0
         )
 
         /** A Sharpen Kernel. This [Matrix3D] can be used as [kernel] for [Convolute3Filter] */
         val KERNEL_SHARPEN: Matrix3D = Matrix3D.fromRows3x3(
-            0f, -1f, 0f,
-            -1f, +4f, -1f,
-            0f, -1f, 0f
+            -1.0, -1.0, -1.0,
+            -1.0, +9.0, -1.0,
+            -1.0, -1.0, -1.0
```

Comment:
Where did you get those numbers?
https://en.wikipedia.org/wiki/Kernel_(image_processing)

Reply:
4 is wrong, but should be:
```
0, -1, 0,
-1, +5, -1,
-1, 0, -1
```

Reply:
Here
http://www.roborealm.com/help/Convolution.php

Reply:
Both matrix (with +5 and +9 in the center) works, but the one with +9 in the center shows a better result (the resulting image is sharper), so I decided to write it instead of the matrix with +5.